### PR TITLE
Patch/transaction parameters

### PR DESCRIPTION
**Pull request by jubril-paystack**
- This PR adds missing transaction parameter fields
- Add missing payment methods (`eft` and `bank_transfer`)
- Handles an edge case where transaction id is returned as either a string or an integer

### CODE REVIEW

File: CheckoutSDK/Classes/CheckoutViewController.swift [hunk 1]
```diff
@@ -133,7 +133,8 @@ public class CheckoutViewController: UIViewController {
     
     private func parseTransactionResponse(dict: [String : Any]) -> TransactionResponse?  {
         if let data = dict["data"] as? [String : Any] {
-            guard let id = data["trans"] as? String, let reference = data["reference"] as? String else { return nil}
+            guard let reference = data["reference"] as? String else {return nil}
+            let id = data["trans"] as? String ??  String(data["trans"] as? Int ?? 0)
```
Comment (ahmed-paystack) [inline CheckoutSDK/Classes/CheckoutViewController.swift:137]: The previous implementation was terminating the flow if there was no "trans". Can you please double-check if it's ok to pass "0" as default if we don't get this param?

File: CheckoutSDK/Classes/TransactionParams.swift [hunk 2]
```diff
@@ -14,25 +14,28 @@ public struct TransactionParams {
...
     public var plan: String?
     public var invoiceLimit: Int?
     public var subAccount: String?
     public var transactionCharge: Int?
     public var bearer: String?
+    public var metaData: String?
+    public var quantity: Int?
+    public var splitCode: String?
+    public var label: String?
     public var currency: Currency?
```
Comment (ravi-paystack) [inline CheckoutSDK/Classes/TransactionParams.swift:30]: Should this be an optional parameter ? Can we have a transaction without a currency ?

File: CheckoutSDK/Classes/CheckoutViewController.swift (hunk 1 above)
Comment (jubril-paystack) [reply CheckoutSDK/Classes/CheckoutViewController.swift:137]: For context,  I noticed that depending on the payment channel the backend returns `trans` (the transaction ID) as either a string or an integer. The transaction id is set to 0 if no transaction id is set. Ideally there should always be a transaction id returned.

File: CheckoutSDK/Classes/TransactionParams.swift (hunk 2 above)
Comment (jubril-paystack) [reply CheckoutSDK/Classes/TransactionParams.swift:30]: Yeah, currency is is optional. If no currency is set, the backend uses the business' default currency

**Review by ravi-paystack — APPROVED** [review]
@jubril-paystack i have left few comments. I feel this is an urgent PR but these are things we can try to improve slowly in all PRs so that we can incrementally improve SDK.

**Review by Peter-John-paystack — APPROVED** [review]
Yeah im with @ravi-paystack on this, the PR could be improved but since its urgent.

**Review by ahmed-paystack — APPROVED** [review]

**Pull request merged**